go/vt/vttablet/tabletmanager/rpc_external_reparent.go

```diff
@@ -151,10 +152,17 @@ func (agent *ActionAgent) finalizeTabletExternallyReparented(ctx context.Context
 		defer wg.Done()
 		log.Infof("finalizeTabletExternallyReparented: updating tablet record for new master: %v", agent.TabletAlias)
 		// Update our own record to master if needed
+		// We are calling UpdateTabletFields and not ChangeType here because
+		// we need to check the type saved in topo before changing it.
+		// Only if the type is changing do we want to set masterTermStartTime, otherwise
```

Comment:
Even if the type was already MASTER, I think we still want to update the MasterTermStartTime. If the value is newer, it means we have more recently been informed authoritatively that this is the master, so that's the value we're going to use for comparison against other masters and we should persist it in topo.

That means it should be fine to call `ChangeType` here. The only difference is that `ChangeType` doesn't currently have the optimization of returning `NoUpdateNeeded`, but that shouldn't have any semantic effect. We could either leave it unoptimized, or add the optimization to `ChangeType` so it applies everywhere.